Kconfig.zephyr

```diff
@@ -317,6 +317,19 @@ config BUILD_OUTPUT_BIN
 	  Build a "raw" binary zephyr/zephyr.bin in the build directory.
 	  The name of this file can be customized with CONFIG_KERNEL_BIN_NAME.
 
+config BUILD_OUTPUT_EFI
+	bool "Build as an EFI application"
+	default n
+	help
```

Comment:
Should we have this depend on x86_64 right now?  Not sure if we support EFI on any other arch's?